### PR TITLE
fix: modify a programming error according to SM2 Digital Signature ge…

### PR DESCRIPTION
**Pull request by ymj-uos**
…neration algorithm process in clause 6.1 and 6.2 of GMT 0003.2-2012.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/sm2/sm2_sign.c [hunk 1]
```diff
@@ -239,6 +239,15 @@ static ECDSA_SIG *sm2_sig_gen(const EC_KEY *key, const BIGNUM *e)
...
     }
 
+    /*
+     * A3: Generate a random number 𝑘∈[1,𝑛−1] using random number generators;
+     * A4: Compute (𝑥1,𝑦1)=[𝑘]𝐺, and convert the type of data 𝑥1 to be integer
+     *     as specified in Clause 4.2.8 of GM/T 0003.1‒2012;
+     * A5: Compute r=(𝑒+𝑥1) mod 𝑛. If 𝑟=0 or 𝑟+𝑘=𝑛, then go to A3;
+     * A6: Compute 𝑠=(1/(1+𝑑𝐴)⋅(𝑘−𝑟⋅𝑑𝐴)) mod 𝑛. If 𝑠=0, then go to A3;
+     * A7: Convert the type of data (𝑟,𝑠) to be bit strings according to the details
+     *     in Clause 4.2.2 of GM/T 0003.1‒2012. Then the signature of message 𝑀 is (𝑟,𝑠).
```
Comment (kaduk) [inline crypto/sm2/sm2_sign.c:249]: #4879 (and #4877) seem to indicate that some compilers used to build openssl do not tolerate non-ASCII characters in the source.

Comment (ymj-uos) [reply crypto/sm2/sm2_sign.c]: have removed non-ASCII characters.

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master and 3.0 branches after squashing the commits and slightly rewording the commit message.

**Comment by t8m** [comment]
Thank you for the contribution.

**Pull request closed without merging**